executor/server.go

```diff
@@ -359,6 +360,7 @@ func (s *Server) connectToEtcdDiscovery(ctx context.Context) (metaStoreSession,
 		etcdCli, adapter.ExecutorInfoKeyAdapter, defaultDiscoverTicker)
 	if old != nil {
 		s.discovery.CopySnapshot(old.SnapshotClone())
+		s.discovery.Close()
```

Comment:
I'm confused with here. Why close new discovery instead of close the old one?

Reply:
addressed, it should close the old one.

Reply:
Then is there a unit-test covering this?

Reply:
Not yet, this function is mocked in unit test.

Reply:
the function `connectToEtcdDiscovery` will be refactored in next PR.

Reply:
it's fine